tools/doc/markdown.js

```diff
@@ -2,18 +2,29 @@
 
 const visit = require('unist-util-visit');
 
+const referenceToLocalMdFile = /^(?![+a-z]+:)([^#?]+)\.md(#.+)?$/i;
+
 module.exports = {
-  replaceLinks
+  replaceLinks,
+  referenceToLocalMdFile,
 };
 
 function replaceLinks({ filename, linksMapper }) {
   return (tree) => {
     const fileHtmlUrls = linksMapper[filename];
 
+    visit(tree, (node) => {
```

Comment:
I have just copied the code below, I haven't used the API docs nor do I know where to find it.

Reply:
From what I was able to find out, [unist-util-visit](https://www.npmjs.com/package/unist-util-visit) uses the [**Uni**versal **S**yntax **T**ree](https://github.com/syntax-tree/unist).

@wooorm might be able to clue us in on whether there are additional learning resources, but so far this code seems to be fully functional to me. It would be nice to get a better understanding of how this works though.

Reply:
readme for unist-util-visit(-parents) should have enough info. There are also [some recipes on the website of unified](https://unifiedjs.com/learn/recipe/).